stan/math/prim/prob/inv_wishart_lpdf.hpp

```diff
@@ -87,8 +89,7 @@ return_type_t<T_y, T_dof, T_scale> inv_wishart_lpdf(const T_y& W,
     //      &S(0), S.size(), 1);
     //    lp -= 0.5 * dot_product(S_vec, W_inv_vec); // trace(S * W^-1)
     Eigen::Matrix<return_type_t<T_y, T_scale>, Eigen::Dynamic, Eigen::Dynamic>
-        Winv_S(mdivide_left_ldlt(
-            ldlt_W, S_ref.template selfadjointView<Eigen::Lower>()));
+        Winv_S(mdivide_left_ldlt(ldlt_W, S_ref));
```

Comment:
Why did the `.template selfadjointView<Eigen::Lower>()` get deleted?

Reply:
It was giving me errors so I deleted it. Maybe a `var_value<T>` thing? I'm not sure.

Then I realized that none of the wishart/inverse wishart functions were checking their arguments for symmetry -- they were just using the lower triangle of the input argument (or assuming things were symmetric). I just ended up changing how all that worked. Maybe that should go in a different pull.

Reply:
tbh idk if it's more expensive to check that the thing is symmetric and use `selfAdjointView<>` or if it's just better to pass the whole matrix here. I'd guess it depends on how it's used. I'll leave that call to you. If we drop the need for symmetry here then you can remove that adjoint view. But if we need to check that anyway then self adjoint view will be faster since your telling eigen that matrix is the same across the lower/upper triangular

Reply:
It's more expensive this way. THe other way is kinda weird (pass an asymmetric matrix and it'll just silently use the bottom half and work).

Reply:
Addressed separately here: https://github.com/stan-dev/math/pull/2281